### DISCUSSION TITLE
FRR vlan implementation and STP support

### DISCUSSION THREAD

**Discussion by jbemmel** [ipspace/netlab#1371] [Features]
The Cumulus implementation of vlans uses a single VLAN-aware Linux bridge that runs STP. FRR containers create a bridge per vlan for which stp is off, and hence they cannot be used in topologies containing loops

Should we add a flag to the vlan module to enable STP, or a separate ```stp``` module? Or do we prefer "use Cumulus if you need STP support"?

**Comment by ipspace** [comment]
I would document this as a caveat and move on. I wouldn't expect many people to use FRR as a layer-2 switch, and not having STP with VXLAN/EVPN only speeds up the lab start time (no need to wait 30 seconds).

Also: https://serverfault.com/questions/824621/linux-bridges-vlans-and-rstp -- I don't want to be anywhere near that quagmire.

**Comment by ipspace** [comment]
Implemented in #1373
